### PR TITLE
Update gloo to 1.0.0

### PR DESCRIPTION
**Pull request by StrongMonkey**

### CODE REVIEW

**Comment by StrongMonkey** [comment]
Need to wait for https://github.com/solo-io/gloo/issues/1716 to be solved. Currently running on a custom version of gloo.

**Comment by StrongMonkey** [comment]
Also need to add migration path for upgrading.

**Comment by daxmc99** [comment]
Is this a non-breaking upgrade?

**Comment by StrongMonkey** [comment]
This is actually a breaking change upgrade in gloo, but we takes care of gloo upgrade so there shouldn't need any manual operations. Doesn't guarantee zero downtime.

File: modules/gloo/features/feature.go [hunk 1]
```diff
@@ -38,7 +55,57 @@ func Register(ctx context.Context, rContext *types.Context) error {
...
+			// handle upgrade
+			return upgradeOldVirtualService(rContext.Namespace, rContext.RestConfig)
+		},
 	}
 
 	return feature.Register()
 }
+
+// upgradeOldVirtualService handles compatibility from upgrading gloo from v0.20.x to v1.0.0 for virtualservices.
+func upgradeOldVirtualService(systemNamespace string, config *rest.Config) error {
```
Comment (cbron) [inline modules/gloo/features/feature.go]: This says upgrade, but really it just deletes them so they can be recreated ?

Comment (StrongMonkey) [reply modules/gloo/features/feature.go]: Corrrect, delete non-compatible types

**Pull request merged**